[GlobalISel] Bail out early for big-endian

davemgreen opened the pull request:
If we continue through the function we can currently hit crashes. We can bail out early and fall back to SDAG.

Fixes #103032

arsenm commented on llvm/test/CodeGen/AArch64/GlobalISel/endian_fallback.ll:
```diff
@@ -0,0 +1,18 @@
+; RUN: llc -mtriple aarch64-unknown-linux-musl -O0 < %s | FileCheck %s --check-prefix=CHECK-LE
+; RUN: llc -mtriple aarch64_be-unknown-linux-musl -O0 < %s | FileCheck %s --check-prefix=CHECK-BE
+
+; Make sure we fall-back to SDAG for BE targets.
```
This doesn't really do that unless you check for the fallback remark

davemgreen replied:
Sorry - I missed your remark earlier, my bad. It was crashing earlier so was fairly obvious that it was "fixed". Let me add a check for the remark too.

tschuett commented:
Silly question: why was "unable to translate in big endian mode" not visible in the error report?

davemgreen commented:
It looks like this is failing on a couple of Arm tests because it is expecting the ir-translator to produce testable results even under BE.

> Silly question: why was "unable to translate in big endian mode" not visible in the error report?

By "error report", do you mean the fall-back remarks? -O0 will use -global-isel-abort=0 (I think), so not report the remarks or error out, it just prints a dbgs() message and otherwise silently falls back to SDAG.

tschuett commented:
> It looks like this is failing on a couple of Arm tests because it is expecting the ir-translator to produce testable results even under BE.
> 
> > Silly question: why was "unable to translate in big endian mode" not visible in the error report?
> 
> By "error report", do you mean the fall-back remarks? -O0 will use -global-isel-abort=0 (I think), so not report the remarks or error out, it just prints a dbgs() message and otherwise silently falls back to SDAG.

The message was not visible in https://github.com/llvm/llvm-project/issues/103032. You put a return after the remark. I would have expected the remark to visible.

davemgreen commented:
Yeah it will depend on the value of -global-isel-abort, which I believe is =0 when enabled through -O0. https://godbolt.org/z/49Pav884n

davemgreen commented:
I've added an option for testing the Arm backend under BE.

davemgreen committed:
[GlobalISel] Bail out early for big-endian

If we continue through the function we can currently hit crashes. We can bail
out early and fall back to SDAG.

davemgreen committed:
Check for fallback remark
